### DISCUSSION TITLE
Sagepay URL Changes

### DISCUSSION THREAD

**Discussion by SPB105438** [opencart/opencart#13760] [General]
Just for your information - I know you have been updating the sagepay modules recently.

Following the updates last year with the Opayo gateway becoming a payment product under the Elavon brand, we wrote to you regarding upcoming changes to the URLs you use for your transactions and reporting.  

We wanted to remind you that these URLs must be updated by 31st March 2024.  

We've already added these updated URLs to our [Developer Hub](https://u19570232.ct.sendgrid.net/ls/click?upn=u001.pwiEQBzkVVV-2F8MFjnFKkh2bClMn9nVkxanXKTPnENRUeoXchwYZYzHF2L2idcpD3O-XI_u8uPQAx6kRPID3-2FlDYfRwjNMoIOTzeBUn9LzvWdTmo-2BCc9m7RQGX1AFuM0CZwLSp2OI4qQrm12jmvpkLl9q7i9bQ9aboFZspO-2BC83BRGdIQqakSh1spseYiR-2F7cFmp746xQ2js3ZTuuabRJnT3pwCZZhXGGXwjBBA-2FFtenOfNgPGk8-2FmaqbW7heKVsyPR-2F-2F7ml8Sqexl1RcaduikRonG4vhafXA7o4Ol801AtUsDVhVk7GGdAGH4jY7O-2BPiZqZc1WjfPqZbtZq6qr9rP0LlB-2BpDWfShaVxGZMQxbXseZP-2BHVKbBHcOX-2FxvBBl9UZ8RdNeiIlRiobx8X0YaRQmWk9LS-2FNmXqtQa46qKauZZPt1hqIc7Bc-2FAoIOcUSsL6-2Ft3q5tjywuWSnJ0OKCYbElQTxmSTeQ0FTx7ZP0EEJxkpVU-2F-2FGrdNZwI0k6yc8nHjiLKXKI1anvnuqKa-2BOWcrujpegXUGaRzVfyrpnZVx2QxHvl1LSfSB0gBfB5JhEu0Ej2-2F79JW7PcCJPlPomuPZbaEuqhGDSfuWGYtR-2FohEoYKOPyozoaiJIhDciXHUe0wflfPdPsJmcMtn-2FwgY0Xtr2Ua2pQ4flWtm4GEMNPqBhjA1vk9XhfqxiYFN-2BrmgtjJ6vl4JpQm87NJo35Ulaf4R4-2FAf9dluCY77-2BbmHGZYdAW0WOoSQwWKegBIHhayVuuXz1Bl5Eu5FrpGndvbGZ91A4jkvR9SWRgpXb8uZ6cMS5-2FWWA7IEj-2F-2FVSm26ejTrISVCgiXurnQ3xbOQIbA6SW2Fer8rcMPnaIUBYk28HJgx67z8d6wBWtFlN0aMRso1l6YTC-2BmRyDEj-2Fe0D6sxLyWCb77b-2FFQuZluQ8zMxiNJTMKqZUNEK26UhbJ-2FTSsKBAZ1uCFgEa1fNIaNN0Ew7xMOEytMbzYC7FceHmoTaoATYFo6p33uKdUW7JkbZ9yaDHxpauD9mEQczeZVLk8ngyd3cSdKNsDdJlZ0iZsDDWAeJ28ZaxqSn8MVQ1hje0EXZ3-2BFADgu9IGQ13tI0kqF-2BPf2raL-2BJ7nD1Bk066F7HS-2FZMbJMA4KbCQ-3D) to help you make the changes, you can find the URLs specific to your integration below:

[Pi integration](https://u19570232.ct.sendgrid.net/ls/click?upn=u001.pwiEQBzkVVV-2F8MFjnFKkh2bClMn9nVkxanXKTPnENRXQcQ-2FQGB6OdZ7y8N2ptHod4O6NMeGnc-2B6h5mikhYAbLGNJ7MsB-2FPkshjApb7ZqCIs-3DniAR_u8uPQAx6kRPID3-2FlDYfRwjNMoIOTzeBUn9LzvWdTmo-2BCc9m7RQGX1AFuM0CZwLSp2OI4qQrm12jmvpkLl9q7i9bQ9aboFZspO-2BC83BRGdIQqakSh1spseYiR-2F7cFmp746xQ2js3ZTuuabRJnT3pwCZZhXGGXwjBBA-2FFtenOfNgPGk8-2FmaqbW7heKVsyPR-2F-2F7ml8Sqexl1RcaduikRonG4vhafXA7o4Ol801AtUsDVhVk7GGdAGH4jY7O-2BPiZqZc1WjfPqZbtZq6qr9rP0LlB-2BpDWfShaVxGZMQxbXseZP-2BHVKbBHcOX-2FxvBBl9UZ8RdNeiIlRiobx8X0YaRQmWk9LS-2FNmXqtQa46qKauZZPt1hqIc7Bc-2FAoIOcUSsL6-2Ft3q5tjywuWSnJ0OKCYbElQTxmSTeQ0FTx7ZP0EEJxkpVU-2F-2FGrdNZwI0k6yc8nHjiLKXKI1anvnuqKa-2BOWcrujpegXUGaRzVfyrpnZVx2QxHvl1LSfSB0gBfB5JhEu0Ej2-2F79JW7PcCJPlPomuPZbaEuqhGDSfuWGYtR-2FohEoYKOPyozoaiJIhDciXHUe0wflfPdPsJmcMtn-2FwgY0Xtr2Ua2pQ4flWtm4GEMNPqBhjA1vk9XhfqxiYFN-2BrmgtjJ6vl4JpQm87NJo35Ulaf4R4-2FAf9dluCY77-2BbmHGZYdAW0WOoSQwWKegBIHhayVuuXz1Bl5Eu5FrpGndvbGZ91A4jkvR9SWRgpXb8uZ6cMS5-2FWWA7IEj-2F-2FVSm26ejTrISVCgiXurnQ3xbOQIbA6SW2Fer8rcMPnaIUBYk28HJgx67z8d6wBWtFlN0aMRso1l6YTC-2BmRyDEj-2Fe0D6sxLyWCb77b-2FFQuZluQ8zMxiNJTMKqZUNEK26UhbJ-2FTSsKBAZ1uCFgEa1FVlSOiK7BxJVIC-2BbVb6U1aFcOJubAj-2FIBGSq45h7z8oqpON5UfvZGOGY3oOCvzn0NBaR-2B4bZmAZtB0oiLW2I77dO4TiEQ66JFZ541QHWGU8p6uOv9g2AkO9fQPfwfnQS8SkXfhHgGqbvstc9gfpqA5V1H9TovtSCLux7kQAB5VE-3D)
[Direct integration](https://u19570232.ct.sendgrid.net/ls/click?upn=u001.pwiEQBzkVVV-2F8MFjnFKkh2bClMn9nVkxanXKTPnENRXQcQ-2FQGB6OdZ7y8N2ptHod-2BtIYfJpssELk4TDg2C1N5bvgs3rq3H5FI2yD5nY9JL2MfW7DpyqMB7y5aZUntldE4wwe_u8uPQAx6kRPID3-2FlDYfRwjNMoIOTzeBUn9LzvWdTmo-2BCc9m7RQGX1AFuM0CZwLSp2OI4qQrm12jmvpkLl9q7i9bQ9aboFZspO-2BC83BRGdIQqakSh1spseYiR-2F7cFmp746xQ2js3ZTuuabRJnT3pwCZZhXGGXwjBBA-2FFtenOfNgPGk8-2FmaqbW7heKVsyPR-2F-2F7ml8Sqexl1RcaduikRonG4vhafXA7o4Ol801AtUsDVhVk7GGdAGH4jY7O-2BPiZqZc1WjfPqZbtZq6qr9rP0LlB-2BpDWfShaVxGZMQxbXseZP-2BHVKbBHcOX-2FxvBBl9UZ8RdNeiIlRiobx8X0YaRQmWk9LS-2FNmXqtQa46qKauZZPt1hqIc7Bc-2FAoIOcUSsL6-2Ft3q5tjywuWSnJ0OKCYbElQTxmSTeQ0FTx7ZP0EEJxkpVU-2F-2FGrdNZwI0k6yc8nHjiLKXKI1anvnuqKa-2BOWcrujpegXUGaRzVfyrpnZVx2QxHvl1LSfSB0gBfB5JhEu0Ej2-2F79JW7PcCJPlPomuPZbaEuqhGDSfuWGYtR-2FohEoYKOPyozoaiJIhDciXHUe0wflfPdPsJmcMtn-2FwgY0Xtr2Ua2pQ4flWtm4GEMNPqBhjA1vk9XhfqxiYFN-2BrmgtjJ6vl4JpQm87NJo35Ulaf4R4-2FAf9dluCY77-2BbmHGZYdAW0WOoSQwWKegBIHhayVuuXz1Bl5Eu5FrpGndvbGZ91A4jkvR9SWRgpXb8uZ6cMS5-2FWWA7IEj-2F-2FVSm26ejTrISVCgiXurnQ3xbOQIbA6SW2Fer8rcMPnaIUBYk28HJgx67z8d6wBWtFlN0aMRso1l6YTC-2BmRyDEj-2Fe0D6sxLyWCb77b-2FFQuZluQ8zMxiNJTMKqZUNEK26UhbJ-2FTSsKBAZ1uCFgEa1Q-2FKPL8dDrehjLB-2BQ8UBQdTkMoAbcmC4iY0ycVjuczQXS0P-2BbTQbact-2FnDKfYV0cnCViNxXXKK3sXXCzmFZ7iDgYzcmb3TK2T6wArmackDgpqQlST9DXewiD-2FGZq-2FCIYdaxNpKM8jkpvpfTIBU6pSoTPS-2BQiZsoYyzqX-2FPQvLq0g-3D)
[Form integration](https://u19570232.ct.sendgrid.net/ls/click?upn=u001.pwiEQBzkVVV-2F8MFjnFKkh2bClMn9nVkxanXKTPnENRXQcQ-2FQGB6OdZ7y8N2ptHodAzNjdsK3YoTwYdvMrkpuvH29EQG8bJl7S4wYQsD65FOma3QbAGzIyDZqiN8grT9QGJYz_u8uPQAx6kRPID3-2FlDYfRwjNMoIOTzeBUn9LzvWdTmo-2BCc9m7RQGX1AFuM0CZwLSp2OI4qQrm12jmvpkLl9q7i9bQ9aboFZspO-2BC83BRGdIQqakSh1spseYiR-2F7cFmp746xQ2js3ZTuuabRJnT3pwCZZhXGGXwjBBA-2FFtenOfNgPGk8-2FmaqbW7heKVsyPR-2F-2F7ml8Sqexl1RcaduikRonG4vhafXA7o4Ol801AtUsDVhVk7GGdAGH4jY7O-2BPiZqZc1WjfPqZbtZq6qr9rP0LlB-2BpDWfShaVxGZMQxbXseZP-2BHVKbBHcOX-2FxvBBl9UZ8RdNeiIlRiobx8X0YaRQmWk9LS-2FNmXqtQa46qKauZZPt1hqIc7Bc-2FAoIOcUSsL6-2Ft3q5tjywuWSnJ0OKCYbElQTxmSTeQ0FTx7ZP0EEJxkpVU-2F-2FGrdNZwI0k6yc8nHjiLKXKI1anvnuqKa-2BOWcrujpegXUGaRzVfyrpnZVx2QxHvl1LSfSB0gBfB5JhEu0Ej2-2F79JW7PcCJPlPomuPZbaEuqhGDSfuWGYtR-2FohEoYKOPyozoaiJIhDciXHUe0wflfPdPsJmcMtn-2FwgY0Xtr2Ua2pQ4flWtm4GEMNPqBhjA1vk9XhfqxiYFN-2BrmgtjJ6vl4JpQm87NJo35Ulaf4R4-2FAf9dluCY77-2BbmHGZYdAW0WOoSQwWKegBIHhayVuuXz1Bl5Eu5FrpGndvbGZ91A4jkvR9SWRgpXb8uZ6cMS5-2FWWA7IEj-2F-2FVSm26ejTrISVCgiXurnQ3xbOQIbA6SW2Fer8rcMPnaIUBYk28HJgx67z8d6wBWtFlN0aMRso1l6YTC-2BmRyDEj-2Fe0D6sxLyWCb77b-2FFQuZluQ8zMxiNJTMKqZUNEK26UhbJ-2FTSsKBAZ1uCFgEa1RqtS-2FNrQNUoxRmXQhE3ol-2FaTw8NcLOj-2FPk37p9G-2Fay6w9NGh5xtvofuxhIStOSYTUiKUxqBa-2Fqotv9NIkd-2FD42L9-2FAqcLTZQCFEaW0SnTjoF-2FT0c-2FKgh-2FiWC-2ByQY5nda0ZMoMnvs83z4bN-2FbYx8HQbVIWlZmczI4XDq7V3c1nJY-3D)
[Server integration](https://u19570232.ct.sendgrid.net/ls/click?upn=u001.pwiEQBzkVVV-2F8MFjnFKkh2bClMn9nVkxanXKTPnENRXQcQ-2FQGB6OdZ7y8N2ptHodgM6SSaZPnkwJQVT04aPZiHCmgqMDf7SM-2BZSKG4DJdhA4n-2BuAyEyeKuTIz4bLW-2FKW2pGw_u8uPQAx6kRPID3-2FlDYfRwjNMoIOTzeBUn9LzvWdTmo-2BCc9m7RQGX1AFuM0CZwLSp2OI4qQrm12jmvpkLl9q7i9bQ9aboFZspO-2BC83BRGdIQqakSh1spseYiR-2F7cFmp746xQ2js3ZTuuabRJnT3pwCZZhXGGXwjBBA-2FFtenOfNgPGk8-2FmaqbW7heKVsyPR-2F-2F7ml8Sqexl1RcaduikRonG4vhafXA7o4Ol801AtUsDVhVk7GGdAGH4jY7O-2BPiZqZc1WjfPqZbtZq6qr9rP0LlB-2BpDWfShaVxGZMQxbXseZP-2BHVKbBHcOX-2FxvBBl9UZ8RdNeiIlRiobx8X0YaRQmWk9LS-2FNmXqtQa46qKauZZPt1hqIc7Bc-2FAoIOcUSsL6-2Ft3q5tjywuWSnJ0OKCYbElQTxmSTeQ0FTx7ZP0EEJxkpVU-2F-2FGrdNZwI0k6yc8nHjiLKXKI1anvnuqKa-2BOWcrujpegXUGaRzVfyrpnZVx2QxHvl1LSfSB0gBfB5JhEu0Ej2-2F79JW7PcCJPlPomuPZbaEuqhGDSfuWGYtR-2FohEoYKOPyozoaiJIhDciXHUe0wflfPdPsJmcMtn-2FwgY0Xtr2Ua2pQ4flWtm4GEMNPqBhjA1vk9XhfqxiYFN-2BrmgtjJ6vl4JpQm87NJo35Ulaf4R4-2FAf9dluCY77-2BbmHGZYdAW0WOoSQwWKegBIHhayVuuXz1Bl5Eu5FrpGndvbGZ91A4jkvR9SWRgpXb8uZ6cMS5-2FWWA7IEj-2F-2FVSm26ejTrISVCgiXurnQ3xbOQIbA6SW2Fer8rcMPnaIUBYk28HJgx67z8d6wBWtFlN0aMRso1l6YTC-2BmRyDEj-2Fe0D6sxLyWCb77b-2FFQuZluQ8zMxiNJTMKqZUNEK26UhbJ-2FTSsKBAZ1uCFgEa1Y0dAP8kKlz81vCu1r8SM2HKzi0138TAErVQGG73Yf31ZTcvnM1RrLyQOj7WQpxszsdSOovlMCDZ9YlE1O8xzmW-2BVgwIogcO3PsbFnEoIyZgdYpFI5pQtgSr6L0vOIvWeVp9cmXk0Q7lsWzV9VyKnIVv5jC0qExYj85g9rwoJjMg-3D)
[RAAPI](https://u19570232.ct.sendgrid.net/ls/click?upn=u001.pwiEQBzkVVV-2F8MFjnFKkh2bClMn9nVkxanXKTPnENRXQcQ-2FQGB6OdZ7y8N2ptHodJr4SU-2Bhb13t5OB1yyxvJ52oQcirZMqg-2BuPcrvsREQTw-3D4lUU_u8uPQAx6kRPID3-2FlDYfRwjNMoIOTzeBUn9LzvWdTmo-2BCc9m7RQGX1AFuM0CZwLSp2OI4qQrm12jmvpkLl9q7i9bQ9aboFZspO-2BC83BRGdIQqakSh1spseYiR-2F7cFmp746xQ2js3ZTuuabRJnT3pwCZZhXGGXwjBBA-2FFtenOfNgPGk8-2FmaqbW7heKVsyPR-2F-2F7ml8Sqexl1RcaduikRonG4vhafXA7o4Ol801AtUsDVhVk7GGdAGH4jY7O-2BPiZqZc1WjfPqZbtZq6qr9rP0LlB-2BpDWfShaVxGZMQxbXseZP-2BHVKbBHcOX-2FxvBBl9UZ8RdNeiIlRiobx8X0YaRQmWk9LS-2FNmXqtQa46qKauZZPt1hqIc7Bc-2FAoIOcUSsL6-2Ft3q5tjywuWSnJ0OKCYbElQTxmSTeQ0FTx7ZP0EEJxkpVU-2F-2FGrdNZwI0k6yc8nHjiLKXKI1anvnuqKa-2BOWcrujpegXUGaRzVfyrpnZVx2QxHvl1LSfSB0gBfB5JhEu0Ej2-2F79JW7PcCJPlPomuPZbaEuqhGDSfuWGYtR-2FohEoYKOPyozoaiJIhDciXHUe0wflfPdPsJmcMtn-2FwgY0Xtr2Ua2pQ4flWtm4GEMNPqBhjA1vk9XhfqxiYFN-2BrmgtjJ6vl4JpQm87NJo35Ulaf4R4-2FAf9dluCY77-2BbmHGZYdAW0WOoSQwWKegBIHhayVuuXz1Bl5Eu5FrpGndvbGZ91A4jkvR9SWRgpXb8uZ6cMS5-2FWWA7IEj-2F-2FVSm26ejTrISVCgiXurnQ3xbOQIbA6SW2Fer8rcMPnaIUBYk28HJgx67z8d6wBWtFlN0aMRso1l6YTC-2BmRyDEj-2Fe0D6sxLyWCb77b-2FFQuZluQ8zMxiNJTMKqZUNEK26UhbJ-2FTSsKBAZ1uCFgEa1O-2F0HiqnfEM6RnTRfjIL2zBpo31QzNIdmSf-2FaG73Uw-2FR-2BIie-2FxeDd-2Bj-2BWvPOIJhPPwM6ut90bKuGxMpdvq5dh3I-2FF9WHxfQNhElXJ-2BDfH51DdASRtjYeiMXZ0Htje1af-2FWwhYGjZyMm5-2B8Lss1EBQgmfcxetDu-2FdVhcnwYHvbmJA-3D)
[Shared API](https://u19570232.ct.sendgrid.net/ls/click?upn=u001.pwiEQBzkVVV-2F8MFjnFKkh2bClMn9nVkxanXKTPnENRXQcQ-2FQGB6OdZ7y8N2ptHodHcXDaL5B3z4ELtWWnDUJYX5n3w7D6oky4-2FV-2B4JAmA-2BU-3D-O1G_u8uPQAx6kRPID3-2FlDYfRwjNMoIOTzeBUn9LzvWdTmo-2BCc9m7RQGX1AFuM0CZwLSp2OI4qQrm12jmvpkLl9q7i9bQ9aboFZspO-2BC83BRGdIQqakSh1spseYiR-2F7cFmp746xQ2js3ZTuuabRJnT3pwCZZhXGGXwjBBA-2FFtenOfNgPGk8-2FmaqbW7heKVsyPR-2F-2F7ml8Sqexl1RcaduikRonG4vhafXA7o4Ol801AtUsDVhVk7GGdAGH4jY7O-2BPiZqZc1WjfPqZbtZq6qr9rP0LlB-2BpDWfShaVxGZMQxbXseZP-2BHVKbBHcOX-2FxvBBl9UZ8RdNeiIlRiobx8X0YaRQmWk9LS-2FNmXqtQa46qKauZZPt1hqIc7Bc-2FAoIOcUSsL6-2Ft3q5tjywuWSnJ0OKCYbElQTxmSTeQ0FTx7ZP0EEJxkpVU-2F-2FGrdNZwI0k6yc8nHjiLKXKI1anvnuqKa-2BOWcrujpegXUGaRzVfyrpnZVx2QxHvl1LSfSB0gBfB5JhEu0Ej2-2F79JW7PcCJPlPomuPZbaEuqhGDSfuWGYtR-2FohEoYKOPyozoaiJIhDciXHUe0wflfPdPsJmcMtn-2FwgY0Xtr2Ua2pQ4flWtm4GEMNPqBhjA1vk9XhfqxiYFN-2BrmgtjJ6vl4JpQm87NJo35Ulaf4R4-2FAf9dluCY77-2BbmHGZYdAW0WOoSQwWKegBIHhayVuuXz1Bl5Eu5FrpGndvbGZ91A4jkvR9SWRgpXb8uZ6cMS5-2FWWA7IEj-2F-2FVSm26ejTrISVCgiXurnQ3xbOQIbA6SW2Fer8rcMPnaIUBYk28HJgx67z8d6wBWtFlN0aMRso1l6YTC-2BmRyDEj-2Fe0D6sxLyWCb77b-2FFQuZluQ8zMxiNJTMKqZUNEK26UhbJ-2FTSsKBAZ1uCFgEa1z69ZF1NgGZR6TZZZAVzdvQzrwL9v6a87805VNauBCpE3pZgBAuqmz813VPUcPt5jGUUv-2FmEPsc9JDSAg-2BIYraAxuFIbmwlTi8JUf2WqShGlCVojbS4jflYH5wZTQ-2FMx2q27BKMp4yBuqKmBO-2Bw3WZZEBVzmcCFbg91RrcfR6D-2BM-3D)

**Comment by mhcwebdesign** [comment]
Thank you for your information. In the OpenCart 3.0.x.x branch we have 3 flavors of SagePay:

catalog/controller/extension/payment/sagepay_direct.php
catalog/controller/extension/payment/sagepay_server.php
catalog/controller/extension/payment/sagepay_us.php

They all have a method **send** using the following URLs:

sagepay_direct.php:

```
	public function send() {
		.....
		if ($this->config->get('payment_sagepay_direct_test') == 'live') {
			$url = 'https://live.sagepay.com/gateway/service/vspdirect-register.vsp';
			$payment_data['VPSProtocol'] = '3.00';
		} elseif ($this->config->get('payment_sagepay_direct_test') == 'test') {
			$url = 'https://test.sagepay.com/gateway/service/vspdirect-register.vsp';
			$payment_data['VPSProtocol'] = '3.00';
		} elseif ($this->config->get('payment_sagepay_direct_test') == 'sim') {
			$url = 'https://test.sagepay.com/Simulator/VSPDirectGateway.asp';
			$payment_data['VPSProtocol'] = '2.23';
		}
		.....
	}

```

sagepay_server.php

```
	public function send() {
		.....
		if ($this->config->get('payment_sagepay_server_test') == 'live') {
			$url = 'https://live.sagepay.com/gateway/service/vspserver-register.vsp';

			$payment_data['VPSProtocol'] = '3.00';
		} elseif ($this->config->get('payment_sagepay_server_test') == 'test') {
			$url = 'https://test.sagepay.com/gateway/service/vspserver-register.vsp';

			$payment_data['VPSProtocol'] = '3.00';
		} elseif ($this->config->get('payment_sagepay_server_test') == 'sim') {
			$url = 'https://test.sagepay.com/Simulator/VSPServerGateway.asp?Service=VendorRegisterTx';

			$payment_data['VPSProtocol'] = '2.23';
		}
		.....
	}

```

sagepay_us.php

```
	public function send() {
		.....
		$url = 'https://www.sagepayments.net/cgi-bin/eftbankcard.dll?transaction';
		.....
	}

```

Would you please so kind to post the correct URLs for above payment methods?
Or perhaps create a pull request for https://github.com/opencart/opencart/tree/3.0.x.x ?

Thanks

**Reply by SPB105438** [reply]
New URLs detailed at https://developer.elavon.com/products/opayo-direct/v1/testing#test-urls 

**Testing URLs**
REGISTRATION | https://sandbox.opayo.eu.elavon.com/gateway/service/vspdirect-register.vsp
3D-SECURE CALLBACK | https://sandbox.opayo.eu.elavon.com/gateway/service/direct3dcallback.vsp
PAYPAL COMPLETION | https://sandbox.opayo.eu.elavon.com/gateway/service/complete.vsp

**Live URLs**
REGISTRATION | https://live.opayo.eu.elavon.com/gateway/service/vspdirect-register.vsp
3D-SECURE CALLBACK | https://live.opayo.eu.elavon.com/gateway/service/direct3dcallback.vsp
PAYPAL COMPLETION | https://live.opayo.eu.elavon.com/gateway/service/complete.vsp

I have tested the new URLs for 3D-SECURE CALLBACK, all working

**Comment by mhcwebdesign** [comment]
Thanks for this. That clarifies it for **sagepay_direct** and **sagepay_server**. Is the URL for **sagepay_us** still good:
https://www.sagepayments.net/cgi-bin/eftbankcard.dll?transaction
?

**Reply by SPB105438** [reply]
I'm unable to find anything regarding sagepayments.net on sagepay / opayo developer portal.
It looks like sagepayment.net is a payment terminal for paya, they also the following urls, maybe it was sold off to [nuvei.com]?

sage.com
sagepayments.net
sagepayments.com
sagepayment.com
sagepaymentsolutions.com
supportcenteronline.com
eftnet.com
net1payments.com
net1pays.com
veruscs.com
verusfm.com
peachtreepaymentsolutions.com
sagepaymentsFI.com
cornerstonepaymentsystems.com
communitycardservices.com
paya.com

**Comment by mhcwebdesign** [comment]
Will keep the sagepay_us as is, for the time being.

What about the Simulator mode for sagepay_direct or sagepay_server? E.g. the old simulator URL is
https://test.sagepay.com/Simulator/VSPDirectGateway.asp

**Reply by SPB105438** [reply]
I believe the simulator mode is no longer available since the move to protocol v3, nothing found on their dev portal.

**Comment by mhcwebdesign** [comment]
> I believe the simulator mode is no longer available since the move to protocol v3, nothing found on their dev portal.

Yeah, I think you are right. Will remove the simulator mode. 

Another question: What do you about the VPSProtocol? The docs seem to suggest using version 4.0.0 (as is the case with the OpenCart Opayo payment extension), yet the sagepay_direct and sagepay_server still use VPSProtocol 3.0.0, which might not work with the new URLs?

**Reply by SPB105438** [reply]
Checking mine, im still using  sagepay_direct and v3 with the new URLs, everything is working fine, however, i would agree they should be updated to use version 4 per the docs

**Reply by SPB105438** [reply]
https://developer.elavon.com/products/opayo-server/v1/protocol-223--400-migration-guide-4814

**Comment by mhcwebdesign** [comment]
OK, I have now updated the SagePay URLs, and disabled the simulator, for sagepay_direct and sagepay_server, see https://github.com/opencart/opencart/pull/13766
